dspace-rest/src/main/java/org/dspace/rest/CommunitiesResource.java

```diff
@@ -145,7 +145,7 @@ public Community getCommunity(@PathParam("community_id") Integer communityId, @Q
 
         log.info("Reading all communities.(offset=" + offset + " ,limit=" + limit + ").");
         org.dspace.core.Context context = null;
-        ArrayList<Community> communities = null;
+        ArrayList<Community> communities = new ArrayList<Community>();
```

Comment:
This change doesn't do anything. the arraylist is initialized at line155. If you are trying to propogate the expand, that was deliberately omitted from being recursive. Extremely bad performance, if you attempted to pass the expand all to everything. i.e. you could get all top communities, and all of their subcommunities, and their collections and all of their items, and all of their bitstreams. Very cool, and convenient for a small instance, would crush a larger site. Perhaps you could invent a new expand option. Such as ?expand=sudo-all-yes-I-understand-the-implications